syntax = "proto3";

package org.apache.dubbo.samples.proto;

message GreeterRequest {
  string name = 1;
}

message GreeterReply {
  string message = 1;
}

service Greeter{

  // unary
  rpc sayHello(GreeterRequest) returns (GreeterReply);

  // client stream
  rpc clientStream(stream GreeterRequest) returns (GreeterReply);

  // server stream
  rpc serverStream(GreeterRequest) returns (stream GreeterReply);

  // bi-directional stream
  rpc biStream(stream GreeterRequest) returns (stream GreeterReply);


}
